### DISCUSSION TITLE
Connection reset by peer issues

### DISCUSSION THREAD

**Discussion by BillyGriffiths** [go-resty/resty#681] [Q&A] [answered]
I'm sitting with a very frustrating situation that I'm struggling to get to the bottom of. 

I use AWS lambda and Resty to make calls to APIs. I've rate-limited these requests to 1 per second, but keep running into `read: connection reset by peer` issues on certain hosts. 

On some hosts it comes and goes. A couple of requests would return `200`, and then start returning `read: connection reset by peer`, a few minutes later it will return `200` again and the cycle continues. Is there something I'm missing with regards to correctly keeping-alive/closing the connection that would cause this?

**Comment by jeevatkm** [comment, marked as answer]
It seems like server-side concerns. You could create a custom transport and disable the keep-alive feature.
